conans/client/conan_api.py

```diff
@@ -56,35 +58,44 @@
 from conans.model.version import Version
 from conans.model.workspace import Workspace
 from conans.paths import BUILD_INFO, CONANINFO, get_conan_user_home
+from conans.paths.package_layouts.package_cache_layout import PackageCacheLayout
 from conans.search.search import search_recipes
 from conans.tools import set_global_instances
 from conans.unicode import get_cwd
 from conans.util.files import exception_message_safe, mkdir, save_files
 from conans.util.log import configure_logger
 from conans.util.tracer import log_command, log_exception
 
-
 default_manifest_folder = '.conan_manifests'
 
 
 def api_method(f):
-    def wrapper(*args, **kwargs):
-        api = args[0]
-        api.create_app()
-        try:
-            curdir = get_cwd()
-            log_command(f.__name__, kwargs)
-            with tools.environment_append(api.app.cache.config.env_vars):
-                return f(*args, **kwargs)
-        except Exception as exc:
-            msg = exception_message_safe(exc)
+    def wrapper(api, *args, **kwargs):
+        silence = kwargs.pop("silence", False)
```

Comment:
Some functions are taking the `self.user_io` from the ConanAPI (they can be using the output contained there) and not the `self.app.user_io`. 

If I fix that, nothing taken from the `ConanAPI` and everything from the `ConanApp`, then I don't need to modify the `app.xxxx` objects, I would only need to pass _my_ output to the `create_app` function (or the `quiet` argument) and it would be able to use the `app.config` instead of the environment variable.

I think that those are changes worth making